app/models/room_type.rb

```diff
@@ -1,3 +1,42 @@
 class RoomType < ApplicationRecord
   has_many :room_type_facilities, dependent: :destroy
+  has_many :facilities, through: :room_type_facilities, source: :facility
+  has_many :rooms, dependent: :nullify
+
+  scope :latest, ->{order created_at: :desc}
+  scope :desc_price, ->{order price: :desc}
+  scope :asc_price, ->{order price: :asc}
+  scope :filter_by_view_type, lambda{|v|
+                                having("view_type = ?", v) if v.present?
+                              }
+  scope :filter_by_room_type, lambda{|t|
+                                having("name = ?", t) if t.present?
+                              }
```

Comment:
```suggestion
scope :filter_by_room_type, ->(t) { having("name = ?", t) if t.present? }
```
Như này cho gọn đc ko em